pkg/queue/manager.go

```diff
@@ -547,3 +570,36 @@ func (m *Manager) reportPendingWorkloads(cqName string, cq ClusterQueue) {
 	}
 	metrics.ReportPendingWorkloads(cqName, active, inadmissible)
 }
+
+func (m *Manager) Start(ctx context.Context) error {
+	log := ctrl.LoggerFrom(ctx).WithName("clusterQueueSnapshot")
```

Comment:
We should not start any snapshotting if MaxCount is 0

https://github.com/kubernetes-sigs/kueue/tree/main/keps/168-pending-workloads-visibility#in-memory-snapshot-of-the-clusterqueue

Reply:
I left this condition here:
https://github.com/kubernetes-sigs/kueue/pull/1069/commits/09e73cf64435a25f2ff2c743642a2f219ef2030c#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R189